### PR TITLE
feat: added cbr support

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description
Added CBR support

### Git-issue
***https://github.ibm.com/GoldenEye/issues/issues/4833***

#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Review by Aayush-Abhyarthi — APPROVED** [review]

File: examples/complete/main.tf [hunk 1]
```diff
@@ -23,6 +23,37 @@ module "key_protect_all_inclusive" {
...
+  source           = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-zone-module?ref=v1.2.0"
+  name             = "${var.prefix}-VPC-network-zone"
+  zone_description = "CBR Network zone representing VPC"
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  addresses = [{
+    type  = "vpc", # to bind a specific vpc to the zone
+    value = ibm_is_vpc.example_vpc.crn,
+  }]
+}
+
```
Comment (ocofaigh) [inline examples/complete/main.tf:55]: You have created a CBR zone, but you never passed any CBR rule to the event streams module?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Please update the `examples/complete/README.md` file. Update the title to `Complete example with BYOK encryption and CBR rules` and also add a bullet point saying:
- A context-based restriction (CBR) rule to only allow Event Streams to be accessible from within the VPC.

File: examples/complete/main.tf (hunk 1 above)
Comment (Khuzaima05) [reply examples/complete/main.tf:55]: ok i will add.

**Comment by Khuzaima05** [comment]
@ocofaigh Getting this error `CreateRuleWithContext failed Invalid API type 'crn:v1:bluemix:public:context-based-restrictions::::api-type:data-plane'. Service 'messagehub' does not support any API type`.  So should i remove the API type?
As there is no scope to API in event stream service named "messagehub" according to https://cloud.ibm.com/docs/account?topic=account-context-restrictions-whatis#cbr-adopters

**Comment by ocofaigh** [comment]
@jojustin could you please advise on above? ^

File: main.tf [hunk 2]
```diff
@@ -84,3 +84,40 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
...
+        name     = "serviceName"
+        value    = "messagehub"
+        operator = "stringEquals"
+      }
+    ]
+  }]
+  operations = [{
+    api_types = [
+      {
+        api_type_id = "crn:v1:bluemix:public:context-based-restrictions::::api-type:data-plane"
```
Comment (jojustin) [inline main.tf:119]: @Khuzaima05 On the error that you mentioned at https://github.com/terraform-ibm-modules/terraform-ibm-event-streams/pull/28#issuecomment-1564208222 is working as expected.  There are no supported API Types for messagehub service for CBR integration.  You are trying to add an API type when its not supported.  Please refer to the example https://github.com/terraform-ibm-modules/terraform-ibm-cbr/blob/9935daa1d5d70ca7a51601e54a4bfbe2fa1c023c/examples/multizone-rule/main.tf#L146 to provide an empty operations.

Comment (Khuzaima05) [reply main.tf]: resolved

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**